gatsby/richtext.rst

```diff
@@ -0,0 +1,169 @@
+      },
+    },
+
+  And then create a ``static`` folder in the ``src``. 
+  The plugin requires a folder to be used as source to work even when the file nodes are all generated by our source plugin.
+
+Now files and images can be queried with GraphQL.
+
+In http://localhost:8000/___graphql
+
+.. code-block:: none
+
+  allPloneFile(filter: { _backlinks: { eq: $path } }) {
+    edges {
+      node {
+        id
+        title
+        file {
+          filename
+          publicURL
+        }
+        _path
+      }
+    }
+  }
+
+  allPloneImage(filter: { _backlinks: { eq: $path } }) {
+    edges {
+      node {
+        id
+        title
+        image {
+          publicURL
+          childImageSharp {
+            fixed(width: 600) {
+             ...GatsbyImageSharpFixed
+            }
+          }
+        }
+      _path
+      }
+    }
+  }
+
+Files can be directly made downloadable by using the ``publicURL`` attribute.
+For images, we use a different approach altogether.
+They are first optimized and processed by Sharp plugins before they are used.
+Install them with ``npm install gatsby-image gatsby-plugin-sharp gatsby-transformer-sharp``
+
+Add them to ``gatsby-node.js`` by simply: 
+
+.. code-block:: javascript
+
+  plugins: [
+    'gatsby-plugin-sharp',
+    'gatsby-transformer-sharp'
+  ],
+
+
+Now, the images are available to be queried as per the example above.
+
+.. note:: 
+
+  The fixed width used there is ``600`` but this can be changed as per your requirements.
+  The whole range of options for can be found in the `docs <https://www.gatsbyjs.org/packages/gatsby-plugin-sharp/>`_.
+
+
+RichText Component
+------------------
+
+We already know how images and files can be queried with GraphQL.
+To use them along with the HTML content, we use the RichText Component. 
+
+Before we jump into that, let us inspect how HTML content is handled by the plugin.
+
+Exercise
+++++++++
+
+Explore GraphiQL at http://localhost:8000/___graphql and compare the stringified HTML and serialized React version of the text data.
+
+Hints: Try checking the text field of the nodes of type ``PloneDocument``
+
+..  admonition:: Solution
+    :class: toggle
+
+    .. code-block:: none
+      {
+        allPloneDocument {
+          edges {
+            node {
+              id
+              text {
+                data
+                react
+              }
+            }
+          }
+        }
+      }
+
+    Notice that ``node.text.react`` is in serialized form that can be deserialized and used with React.
+
+.. note:: 
+
+  Internally, `react-serialize <https://www.npmjs.com/package/react-serialize>`_ is used by the RichText component to handle serialized HTML data.
+  This eliminates the use of ``dangerouslySetInnerHTML``, which is recommended to be avoided.
+
+
+The usage of the RichText component is fairly simple. 
+Take the case of ``Document.js`` which handles all nodes of type ``PloneDocument`` on page creation.
```

Comment:
Avoid opinionated terms like "simple", as that can frustrate a user who just ain't getting it. "Arg! It says it's simple, so I must be less than simple because I don't understand!"

```
The following is a usage example of the RichText component.
``Document.js`` handles all nodes of type ``PloneDocument`` on page creation.
```